### ISSUE TITLE
all except BaseException should be except Exception

### ISSUE THREAD

**Issue by hellysmile** [aio-libs/aiohttp#2444]
## Long story short

https://github.com/aio-libs/aiohttp/pull/2383 fixed flake8 blind exception, but later for example https://github.com/aio-libs/aiohttp/pull/2441 used `except BaseException`

It should be fixed separately in 2.3 and master branches

## Expected behaviour

everywhere should be `except Exception`

## Actual behaviour

Some parts of code uses `except Exception`, some `except BaseException`

**Comment by fafhrd91** [comment]
For 2441 BaseException makes sense, especially because it just re-raise exception

**Comment by socketpair** [comment]
@asvetlov it is actual again. We need a linter.

See https://github.com/aio-libs/aiohttp/blob/7a324fd46ff7dc9bb0bb1bc5afb326e04cf7cef0/aiohttp/client_reqrep.py#L974 for example. Exception in `close()` prevent, for example, exitting from the application.

**Comment by asvetlov** [comment]
No. 
The code catches an exception, closes the response and reraises the exception again.
It should work pretty fine with `CancelledError`, isn't it?

Exception in `close()` unwinds the stack anyway.

**Comment by socketpair** [comment]
Yes, unwinds, but exception in `.close()` will REPLACE, say, SystemExit exception, does not it ?

**Comment by asvetlov** [comment]
Chicken-egg problem.
I have a feeling that a response should be closed in case of exception.
The question is: how should look like an exception in finalizer.
@socketpair do you have a suggestion?

**Comment by socketpair** [comment]
1. We should track only `except Exception`  (and CancelledError also, if applicable).
2. AFAIK, calling `close()` - just an GC optimisation. In good code, GC should kill such items.
3. Another option - try...except...warning+ignore wrapper over `close()`.

There are plenty of exceptions, which are not errors actually. It's wrong design. Like Stopiteration,systemexit,Cancellederror....We should close something only by ERROR. If someone raises such exceptions it should understand what will happen.

That's my opinion.
